torchchat/usages/openai_api.py

```diff
@@ -486,6 +486,20 @@ def _callback(self, x, *, buffer, done_generating):
         pass
 
 
+def create_openai_api_generator(distributed):
+    """
+    Factory method to create an OpenAiApiGenerator
+    """
+
+    if distributed:
+        # Base class order matters to make sure OpenAiApiGeneratorMixin overrides methods in DistributedGenerator and Generator
+        return type('OpenAiApiGenerator', (OpenAiApiGeneratorMixin, DistributedGenerator), {})
```

Comment:
Nit: we can case or ternary just on the Generator class since the rest is the same